tests/spicy/types/bytes/parse-length-eod.spicy

```diff
@@ -1,5 +1,8 @@
 # @TEST-EXEC: printf 'abc123' | spicy-driver %INPUT >output 2>&1
 # @TEST-EXEC: btest-diff output
+#
+# Ensure type-specific optimization kicks in.
+# @TEST-EXEC: spicyc -p %INPUT | grep -q 'b2 = spicy_rt::extractBytes'
```

Comment:
I don't see how I would test for that, nor am I sure it's worth the trouble, we can't test for every possible wrong use of something.

Reply:
If we used `extractBytes` for `skip` fields this would not be a user error, but us generating incorrect code. I think adding something like the following here should test for that.

```ruby
# Make sure we do not generate extraction code with `extractBytes` for skip
# fields since we want to avoid holding them in memory.
# @TEST-EXEC: spicyc -p foo.spicy -o foo.hlto
# @TEST-EXEC-FAIL: grep -q "spicy_rt::extractBytes" foo.hlto

# @TEST-START-FILE foo.spicy
module Test;

public type X = unit {
    b2 : skip bytes &size=2147483647;
};

# @TEST-END-FILE
```